From cd59ed9836011eda94476115d6edf29ea4cd27bd Mon Sep 17 00:00:00 2001
From: Florian Schmaus <flo@geekplace.eu>
Date: Sat, 6 Jun 2020 13:41:17 +0200
Subject: [PATCH 2/2] Add convert_charrefs field to TagStrip

Newer python versions expect this:

ERROR: Diff process exited with an error
Traceback (most recent call last):
  File "/usr/lib/python3.7/site-packages/htmldiff/entry_point.py", line 116, in diff
    diffed_html = gen_side_by_side(diffed_html)
  File "/usr/lib/python3.7/site-packages/htmldiff/lib.py", line 412, in gen_side_by_side
    left = span_to_whitespace(left_side, 'insert')
  File "/usr/lib/python3.7/site-packages/htmldiff/lib.py", line 383, in span_to_whitespace
    stripped = strip_tags(strip)
  File "/usr/lib/python3.7/site-packages/htmldiff/lib.py", line 65, in strip_tags
    st.feed(html_string)
  File "/usr/lib/python3.7/html/parser.py", line 111, in feed
    self.goahead(0)
  File "/usr/lib/python3.7/html/parser.py", line 139, in goahead
    if self.convert_charrefs and not self.cdata_elem:
AttributeError: 'TagStrip' object has no attribute 'convert_charrefs'
---
 src/htmldiff/lib.py | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/htmldiff/lib.py b/src/htmldiff/lib.py
index a3e3218..b22be95 100644
--- a/src/htmldiff/lib.py
+++ b/src/htmldiff/lib.py
@@ -138,6 +138,7 @@ class TagStrip(HTMLParser):
     def __init__(self):
         self.reset()
         self.fed = []
+        self.convert_charrefs = False
 
     def handle_data(self, s):
         self.fed.append(s)
-- 
2.26.2

